app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,261 @@ def self.display_methods
+      security_group = SecurityGroup.find(s)
+      if security_group.nil?
+        add_flash(_("Security Group no longer exists."), :error)
+      elsif security_group.supports_delete?
+        security_groups_to_delete.push(security_group)
+      else
+        add_flash(_("Couldn't initiate deletion of Security Group \"%{name}\": %{details}") % {
+          :name    => security_group.name,
+          :details => security_group.unsupported_reason(:delete)
+        }, :error)
+      end
+    end
+    process_security_groups(security_groups_to_delete, "destroy") unless security_groups_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "security_group"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Security Group was deleted"))
+      else # or (if we deleted what we were showing) we redirect to the listing
+        javascript_redirect :action => 'show_list', :flash_msg => @flash_array[0][:message]
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("security_group_edit")
+    @security_group = find_by_id_filtered(SecurityGroup, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Edit Security Group \"%{name}\"") % { :name  => @security_group.name},
+      :url  => "/security_group/edit/#{@security_group.id}")
+  end
+
+  def get_checked_security_group_id(params)
+    if params[:id]
+      checked_security_group_id = params[:id]
+    else
+      checked_security_groups = find_checked_items
+      checked_security_group_id = checked_security_groups[0] if checked_security_groups.length == 1
+    end
+    checked_security_group_id
+  end
+
+  def new
+    assert_privileges("security_group_new")
+    @security_group = SecurityGroup.new
+    @in_a_form = true
+
+    @ems_choices = {}
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Openstack::NetworkManager").find_each do |ems|
+      @ems_choices[ems.name] = ems.id
+    end
+
+    @cloud_tenant_choices = {}
+    CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
+
+    drop_breadcrumb(:name => _("Add New Security Group"), :url => "/security_group/new")
+  end
+
+  def update
+    assert_privileges("security_group_edit")
+    @security_group = find_by_id_filtered(SecurityGroup, params[:id])
+    options = form_params
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of Security Group \"%{name}\" was cancelled by the user") % {
+        :name => @security_group.name
+      })
+
+    when "save"
+      if @security_group.supports_update?
+        task_id = @security_group.update_security_group_queue(session[:userid], options)
+        add_flash(_("Security Group update failed: Task start failed: ID [%{id}]") %
+                  {:id => task_id.to_s}, :error) unless task_id.kind_of?(Fixnum)
```

Comment:
same, Fixnum should be Integer now (being fixed in https://github.com/ManageIQ/manageiq/pull/13191 for the rest)